### PR TITLE
Wait for transactions to be mined, and for contracts to be deployed i…

### PR DESCRIPTION
**Pull request by vreff**
…n VRF test scripts [sc-25930]

### CODE REVIEW

**Comment by vreff** [comment]
Note: there are currently issues with a VRF smoke test (https://chainlink-core.slack.com/archives/C0117GGJB6Y/p1652733734190529). These changes do not affect this test.

```
Integration.[It] VRFv2 suite @v2vrf with VRF job randomness is fulfilled
Timed out after 120.001s.
Expected success, but got an error:
    <*errors.errorString | 0xc00ed65c30>: {
        s: "Assertion in callback at /home/runner/work/chainlink/chainlink/integration-tests/smoke/vrfv2_test.go:166 failed:\nExpected\n    <int>: 0\nto be ==\n    <int>: 1",
    }
    Assertion in callback at /home/runner/work/chainlink/chainlink/integration-tests/smoke/vrfv2_test.go:166 failed:
    Expected
        <int>: 0
    to be ==
        <int>: 1
```

File: core/scripts/vrfv2/testnet/main.go [hunk 1]
```diff
@@ -822,7 +847,7 @@ func main() {
 		tx, err := linkToken.TransferAndCall(owner, coordinator.Address(), amount, b)
 		helpers.PanicErr(err)
 		fmt.Println("Funding sub", *subID, "TX", helpers.ExplorerLink(chainID, tx.Hash()))
```
Comment (Nic0S) [inline core/scripts/vrfv2/testnet/main.go:849]: Could these log messages also be moved inside confirmTxMined to avoid duplicating them?

**Review by Nic0S — APPROVED** [review]
LG! one possible suggestion

File: core/scripts/vrfv2/testnet/main.go (hunk 1 above)
Comment (vreff) [reply core/scripts/vrfv2/testnet/main.go:849]: I can do that, although not every TX log is generic. For example:

```
fmt.Println("Funding sub", *subID, "TX", helpers.ExplorerLink(chainID, tx.Hash()))
```

I would get rid of this and put a more generic one inside the helper method:

```
fmt.Println("Executing TX:", helpers.ExplorerLink(chainID, tx.Hash()))
```

Honestly just a matter of team preference.

Comment (makramkd) [reply core/scripts/vrfv2/testnet/main.go:849]: Just an idea, but maybe you can add a variadic string parameter that the confirmTxMined helper will log, prior to logging the `Executing TX` log at the end.

Comment (vreff) [reply core/scripts/vrfv2/testnet/main.go]: fair enough

Comment (vreff) [reply core/scripts/vrfv2/testnet/main.go]:
```
Executing TX 0x7a977b9c715cf40ebb1dd2a49682b3f84d2ffa644a84c6551b9eb7e3424ab61a [Uncompressed public key: 12345678 Oracle address: 0x12345678]
```

**Review by Nic0S — APPROVED** [review]

**Pull request merged**